docs/deploy-sphinx-docs-to-ghpages.md

```diff
@@ -0,0 +1,28 @@
+# deploy-sphinx-docs-to-ghpages.yml
+
+This workflow deploys Sphinx documentation and publishes it using GitHub Pages 
+
+Usage 
+=====
+
+```yaml
+name: Deploy docs
+
+
+on: 
+  workflow_dispatch:
+  push:
+      branches: [master]
+  
+jobs:
+
+  deploy-docs:
```

Comment:
The user will be free to set its own name, triggers, and jobs.

Drop this part, just keep the following (from `uses` on), unindented (top-level, just keep relative indentation).
Concerning "How to use a reusable workflow?" you already have a section in the home page :)